pkg/client/typed/generated/extensions/v1beta1/thirdpartyresource.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	api "k8s.io/kubernetes/pkg/api"
+	v1beta1 "k8s.io/kubernetes/pkg/apis/extensions/v1beta1"
+	watch "k8s.io/kubernetes/pkg/watch"
+)
+
+// ThirdPartyResourcesGetter has a method to return a ThirdPartyResourceInterface.
+// A group's client should implement this interface.
+type ThirdPartyResourcesGetter interface {
+	ThirdPartyResources(namespace string) ThirdPartyResourceInterface
+}
+
+// ThirdPartyResourceInterface has methods to work with ThirdPartyResource resources.
+type ThirdPartyResourceInterface interface {
+	Create(*v1beta1.ThirdPartyResource) (*v1beta1.ThirdPartyResource, error)
+	Update(*v1beta1.ThirdPartyResource) (*v1beta1.ThirdPartyResource, error)
+	Delete(name string, options *api.DeleteOptions) error
```

Comment:
I'm not sure when we should use the versioned XXXOptions. I take a look at the manually written clients, they are using the internal XXXOptions.